### PR TITLE
remove the c style string from RigidBodyConstraint

### PR DESCRIPTION
**Pull request by hongkai-dai**

### CODE REVIEW

File: drake/systems/plants/constraint/RigidBodyConstraint.cpp [hunk 1]
```diff
@@ -1095,19 +1037,11 @@ void RelativeQuatConstraint::name(const double* t, std::vector<std::string> &nam
...
-    char time_str_buffer[30];
-    if(t != nullptr)
-    {
-      sprintf(time_str_buffer,"at time %5.2f",*t);
-      time_str+= std::string(time_str_buffer);
-    }
-    sprintf(cnst_name_str_buffer,"%s relative to %s quaternion constraint %s",this->bodyA_name.c_str(),this->bodyB_name.c_str(),time_str.c_str());
-    std::string cnst_name_str(cnst_name_str_buffer);
+    std::string time_str = this->getTimeString(t);
+    std::string tmp_name = this->bodyA_name+" reltative to "+this->bodyB_name+" quaternion constraint"+time_str;
```
Comment (RussTedrake) [inline drake/systems/plants/constraint/RigidBodyConstraint.cpp]: small typo here "reltative"

**Comment by RussTedrake** [comment]
so much better.  thanks hongkai.  might as well fix the one typo, but otherwise it's ready to accept.  maybe also increase the TIMEOUT on testIK?

**Comment by hongkai-dai** [comment]
Fixed them already. @RussTedrake Thanks for the review. It is impressive that you can catch that typo out of many lines of text modification!

**Comment by RussTedrake** [comment]
:+1:  .  thanks hongkai.  will just wait for the build servers.

**Comment by RussTedrake** [comment]
the timeout didn't resolve drake006 -- can you take a look just to make sure we haven't introduced something bad?

**Comment by hongkai-dai** [comment]
CDash just says timeout, no print out message, and I cannot find the random
seed used for that test. I have tried to run the test for several times on
my machine. They finish in 2 minutes in most of the time. There was one
failure, that SNOPT reports INFO 32.

Is there a way I can find out the random seed used on drake 006?

**Pull request merged**